pkg/kubelet/kubelet.go

```diff
@@ -1499,8 +1531,20 @@ func (kl *Kubelet) GetPodStatus(podFullName string, uid types.UID) (api.PodStatu
 
 	info, err := dockertools.GetDockerPodInfo(kl.dockerClient, spec, podFullName, uid)
 
+	for _, c := range spec.Containers {
+		containerStatus := info[c.Name]
+		containerStatus.Ready = kl.readiness.IsReady(containerStatus)
+		info[c.Name] = containerStatus
+	}
+
 	var podStatus api.PodStatus
 	podStatus.Phase = getPhase(&spec, info)
+	if isPodReady(&spec, info) {
```

Comment:
I'm choosing to set the condition to:
```
[]api.PodCondition{{
	Kind:   api.PodReady,
	Status: api.ConditionNone,
}}
```
if any containers in the pod are not ready.